Tribler/Test/Core/Modules/MetadataStore/test_channel_metadata.py

```diff
@@ -285,6 +266,82 @@ def test_commit_channel_torrent(self):
         self.assertTrue(channel.commit_channel_torrent())
         self.assertFalse(channel.commit_channel_torrent())
 
+    @db_session
+    def test_recursive_commit_channel_torrent(self):
+        status_types = [NEW, UPDATED, TODELETE, COMMITTED]
+
+        def all_status_combinations():
+            result = []
+            for card in range(0, len(status_types) + 1):
+                result.extend(list(combinations(status_types, card)))
+            return result
+
+        def gen_coll(parent, collection_status, contents_statuses, recurse=False):
```

Comment:
Can you just name this `generate_collection`?

Reply:
Ok